src/components/add-journey/SearchButton.tsx

```diff
@@ -1,48 +1,39 @@
-import axios from 'axios';
 import { useTranslations } from 'next-intl';
-import toast from 'react-hot-toast';
-import { useThrottleState } from 'react-relaxed';
 
 import { LoadingSpinner } from '@/components/LoadingSpinner';
 import { useJourneySearchStore } from '@/hooks/useJourneySearchStore';
-import { TRANSPORT_API_URL } from '@/constants';
-import type { Journey } from '@/types/opendata';
+import type { Station } from '@/types/opendata';
+import { useGetJourneys } from './hooks';
 
-export const SearchButton: React.FC = () => {
-  const departureTime = useJourneySearchStore((state) => state.departureTime);
-  const departureStation = useJourneySearchStore((state) => state.departureStation);
-  const arrivalStation = useJourneySearchStore((state) => state.arrivalStation);
-  const setJourneys = useJourneySearchStore((state) => state.setJourneys);
-
-  const [, setLoading, loading] = useThrottleState(false, 150);
+type Props = {
+  departureTime: string;
+  departureStation?: Station;
+  arrivalStation?: Station;
+};
 
+export const SearchButton: React.FC<Props> = ({ departureTime, departureStation, arrivalStation }) => {
+  const { isFetching } = useGetJourneys();
   const t = useTranslations();
 
-  const getJourneys = async () => {
-    try {
-      setLoading(true);
-
-      const { data } = await axios.get<{ connections: Journey[] }>(
-        `${TRANSPORT_API_URL}/connections?from=${departureStation?.name}&to=${arrivalStation?.name}&date=${
-          departureTime.split('T')[0]
-        }&time=${departureTime.split('T')[1]}`
-      );
-
-      setJourneys(data.connections);
-    } catch (error) {
-      toast.error('Unable to load journeys');
-    } finally {
-      setLoading(false);
-    }
-  };
+  const setDepartureTime = useJourneySearchStore((state) => state.setDepartureTime);
+  const setDepartureStation = useJourneySearchStore((state) => state.setDepartureStation);
+  const setArrivalStation = useJourneySearchStore((state) => state.setArrivalStation);
 
   return (
     <div className="mt-1 flex w-full items-end">
       <button
-        onClick={getJourneys}
+        onClick={() => {
+          setDepartureTime(departureTime);
+          if (departureStation) {
+            setDepartureStation(departureStation);
+          }
+          if (arrivalStation) {
+            setArrivalStation(arrivalStation);
+          }
+        }}
```

Comment:
changing the responsibilities of the `SearchButton` to set the global state instead of calling the fetch API to make state more reactive